src/main/java/teammates/ui/webapi/action/GetSessionResultsAction.java

```diff
@@ -37,8 +38,12 @@ public void checkSpecificAccessControl() {
         Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
         switch (intent) {
         case INSTRUCTOR_RESULT:
-            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
-            gateKeeper.verifyAccessible(instructor, fs);
+            if (userInfo == null) {
+                throw new UnauthorizedAccessException("Instructor account is required to access this resource");
+            } else {
+                InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+                gateKeeper.verifyAccessible(instructor, fs);
+            }
```

Comment:
Might be worthwhile to also factor this out into a method `getInstructor` to be consistent with `getStudent` below.

Reply:
Sure.